sqs/src/main/scala/akka/stream/alpakka/sqs/scaladsl/SqsFlow.scala

```diff
@@ -18,6 +21,34 @@ object SqsFlow {
       implicit sqsClient: AmazonSQSAsync
   ): Flow[String, Result, NotUsed] =
     Flow.fromGraph(new SqsFlowStage(queueUrl, sqsClient)).mapAsync(settings.maxInFlight)(identity)
+
+  def grouped(queueUrl: String, batchSize: Int, settings: SqsSinkSettings = SqsSinkSettings.Defaults)(
+      implicit sqsClient: AmazonSQSAsync
+  ): Flow[String, Result, NotUsed] = {
+
+    val graph = GraphDSL.create() { implicit builder =>
+      import GraphDSL.Implicits._
+      val groupingStage: FlowShape[String, Seq[String]] = builder.add(Flow[String].grouped(batchSize))
+
+      val sqsStage: FlowShape[Seq[String], Future[List[Result]]] =
+        builder.add(new SqsBatchFlowStage(queueUrl, sqsClient))
+
+      val flattenFutures: FlowShape[Future[List[Result]], List[Result]] =
+        builder.add(Flow[Future[List[Result]]].mapAsync(settings.maxInFlight)(identity))
+
+      val flattenResults: FlowShape[List[Result], Result] = builder.add(Flow[List[Result]].mapConcat(identity))
+      groupingStage ~> sqsStage ~> flattenFutures ~> flattenResults
+
+      FlowShape(groupingStage.in, flattenResults.out)
+    }
+
+    Flow.fromGraph(graph)
+  }
```

Comment:
How about using groupedWithin instead since this will just hold the messages until the batchSize is reached